test/statement_types.py

```diff
@@ -0,0 +1,451 @@
+import numbers
+import os
+
+from sig_utils import overload_scalar, get_cpp_type, arg_types
+
+class CppStatement:
+    """Base class for all statements. Implements some default checks. Should not be used directly"""
+    def __init__(self):
+        raise Exception("CppStatement should never be instantiated directly")
+
+    def is_reverse_mode(self):
+        """By default, a statement is not reverse mode"""
+        return False
+    
+    def is_matrix_like(self):
+        """By default, a statement is not matrix-like"""
+        return False
+
+    def is_varmat_compatible(self):
+        """By default, a statement is not compatible with varmat types"""
+        return False
+
+    def is_expression(self):
+        """By default, a statement is not an expression"""
+        return False
+
+class IntVariable(CppStatement):
+    """Represents an integer variable"""
+    def __init__(self, name, value = None):
+        """
+        Initialize matrix
+        
+        :param overload: Type of overload as string (Prim/Fwd/Rev/etc.)
+        :param name: C++ name of variable
+        :param stan_arg: Stanc3 type string
+        :param value: Scalar value to initialize matrix with
+        """
+        self.name = name
+        if value is None:
+            self.value = 1
+        else:
+            self.value = int(value)
+
+    def cpp(self):
+        """Generate C++"""
+        return "int {name} = {value};".format(name = self.name, value = self.value)
+
+class RealVariable(CppStatement):
```

Comment:
You could reduce the amount of code by merging int and real scalars into a single class.

Reply:
It's mildly more convenient to keep the int casts in one spot.